Skip images without a serialized Swift AST when collecting header

adrian-prantl opened the pull request:
search options for the scratch context.

This is both a performance optimization and as a side effect can avoid
spurious warning messages.

adrian-prantl commented:
@swift-ci test and merge

jimingham commented on source/Symbol/SwiftASTContext.cpp:
```diff
@@ -1037,6 +1037,16 @@ static bool DeserializeAllCompilerFlags(SwiftASTContext &swift_ast,
   return false;
 }
 
+/// Return whether this module contains any serializes Swift ASTs.
```
serialized

adrian-prantl replied:
This is the normal behavior:
```
(lldb) expr -l swift -- func $foo() { print("I wonder what this will do?") }
(lldb) expr -l swift -- $foo
() -> () $R0 = 0x00000001012d6000
```

This is what we get in fallback mode:

```
(lldb) expr -l swift -- func $foo() { print("I wonder what this will do?") }
(lldb) expr -l swift -- $foo() 
error: Couldn't lookup symbols:
  __lldb_expr_4.$foo() -> ()
(lldb) expr -l swift -- 1
(Int) $R4 = 1
(lldb) expr -l swift -- func $foo() { print("I wonder what this will do?") }
(lldb) expr -l swift -- 1
(Int) $R6 = 1
(lldb) expr -l swift -- $foo
error: Couldn't lookup symbols:
  __lldb_expr_7.$foo() -> ()
(lldb) expr -l swift -- $R6
(Int) $R8 = 1
```

This behavior is, however, not affected by this patch. This was already broken.

jimingham reviewed (commented):
What happens if I'm in the fallback mode so I don't have a Target-wide scratch SwiftASTContext, and I stop in a module that returns false from HasSwiftModules and do:

(lldb) expr -l swift -- func $foo() { print("I wonder what this will do?") }

adrian-prantl committed:
Skip images without a serialized Swift AST when collecting header
search options for the scratch context.

This is both a performance optimization and as a side effect can avoid
spurious warning messages.

adrian-prantl committed:
Convert if to early exit (NFC)
